sql/schemachanger: Log events for CREATE/DROP trigger

spilchen opened the pull request:
While debugging an issue, I noticed that CREATE TRIGGER and DROP TRIGGER actions do not generate entries in system.eventlog. The necessary components were mostly in place, but we weren’t calling LogEventForExistingTarget to record the events. This change ensures that trigger creation and deletion are properly logged.

Epic: none
Release note: none

fqazi reviewed (approved):
Nice, find!

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/140969#-:-OIk3aNyDGfN1N2t8koO:b-5gr4us)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

fqazi commented:
@spilchen We probably should backport this?

spilchen commented:
Yeah, it makes sense to backport to 24.3—I believe that's when triggers were introduced.

spilchen committed:
sql/schemachanger: Log events for CREATE/DROP trigger

While debugging an issue, I noticed that CREATE TRIGGER and DROP TRIGGER
actions do not generate entries in system.eventlog. The necessary
components were mostly in place, but we weren’t calling
LogEventForExistingTarget to record the events. This change ensures that
trigger creation and deletion are properly logged.

Epic: none
Release note: none

spilchen commented:
TFTR!

bors r+

rafiss commented:
Nice find!
